client/utils/serde/clubs-and-events.ts

```diff
@@ -0,0 +1,82 @@
+import moment from 'moment-timezone'
+import { Club, Event } from '../../utils'
+
+// TODO: remove these duplicated types from the backend with shared api types
+export interface BasicEventDTO {
+  name: string
+  startTime: string
+  endTime: string
+  bannerImage?: string
+  tags: Array<string>
+}
+
+export interface ClubDTO {
+  name: string
+  description: string
+  links: {
+    bannerImage: string
+    iconImage: string
+    facebook?: string
+    twitter?: string
+    instagram?: string
+    website?: string
+  }
+  tags: Array<string>
+  events: Array<BasicEventDTO>
+}
+
+interface DeserializedClubsAndEvents {
+  deserializedClubs: Array<Club>
+  deserializedEvents: Array<Event>
+}
+
+export const deserializeClubsAndEvents = (
+  serializedClubs: Array<ClubDTO> | undefined
+): DeserializedClubsAndEvents => {
+  if (!serializedClubs) {
+    return {
+      deserializedClubs: [],
+      deserializedEvents: [],
+    }
+  }
+
+  let eventId = 0
+  const deserializedEvents: Array<Event> = []
+
+  const deserializedClubs = serializedClubs.map((club, clubId) => {
+    return {
+      ...club,
+      id: clubId,
+      events: club.events.map((event) => {
+        const deserializedEvent = deserializeEvent(event, eventId++, club)
+        deserializedEvents.push(deserializedEvent)
+
+        return deserializedEvent
+      }),
+    } as Club
+  })
+
+  return {
+    deserializedClubs,
+    deserializedEvents,
+  }
+}
+
+const deserializeEvent = (
+  serializedEvent: BasicEventDTO,
+  eventId: number,
+  club: ClubDTO
+): Event => {
+  const localZone = Intl.DateTimeFormat().resolvedOptions().timeZone
+
+  return {
+    ...serializedEvent,
+    id: eventId,
+    startTime: moment.tz('2022-11-18 10:55', 'Asia/Tokyo').local().tz(localZone),
+    endTime: moment.tz('2022-11-18 10:55', 'Asia/Tokyo').local().tz(localZone),
```

Comment:
Still hard-coding for now? I guess a future todo is to deserialize the actual date.

Reply:
woops, fixed. dates are working correctly :D